Dockerfile.dev

```diff
@@ -0,0 +1,85 @@
+FROM wptd-base
+
+USER root
+
+ARG PB_VERSION=3.4.0
+ARG GO_VERSION=1.8.5
+
+# Used by dev shell scripts
+ENV PB_LIB="/protobuf/src"
+ENV BQ_LIB="/protoc-gen-bq-schema"
+ENV PROTOS="/wptdashboard/protos"
+ENV BQ_OUT="/wptdashboard/bq-schema"
+ENV PY_OUT="/wptdashboard/run/protos"
+
+# Install unzip (for setup), inotify-tools (for dev scripts)
+RUN apt-get update && apt-get install --assume-yes --no-install-suggests \
+    --no-install-recommends \
+    unzip \
+    inotify-tools
+
+# Install python dev dependencies
+RUN pip install \
+    requests==2.18.1 \
+    pycodestyle==2.3.1 \
+    iso8601==0.1.12 \
+    google-cloud==0.26.1 \
+    "protobuf==${PB_VERSION}"
+
+# Install golang
+RUN mkdir /go-fetch
+WORKDIR /go-fetch
+RUN curl -L -o "go.tar.gz" "https://redirector.gvt1.com/edgedl/go/go${GO_VERSION}.linux-amd64.tar.gz" && \
+    tar -C /usr/local -xzf "go.tar.gz" && \
+    cd / && \
+    rm -rf /go-fetch
+
+# Setup go environment
+RUN mkdir /go
+ENV GOPATH=/go
+ENV PATH=$PATH:/usr/local/go/bin:$GOPATH/bin
+
+# Install go dependencies
+RUN go get github.com/golang/lint/golint
+
+# Install protobuf libraries
+RUN mkdir /protobuf-fetch
+WORKDIR /protobuf-fetch
+RUN curl -L -o "protobuf.zip" "https://github.com/google/protobuf/archive/v${PB_VERSION}.zip" && \
+    unzip "protobuf.zip" -d / && \
+    mv "/protobuf-${PB_VERSION}" "/protobuf" && \
+    cd /  && \
+    rm -rf /protobuf-fetch
+
+# Install protobuf compiler
+RUN mkdir /protoc
+WORKDIR /protoc
+RUN curl -L -o "protoc.zip" "https://github.com/google/protobuf/releases/download/v${PB_VERSION}/protoc-${PB_VERSION}-linux-x86_64.zip" && \
+    unzip "protoc.zip" && \
+    cp "bin/protoc" /usr/local/bin/ && \
+    chmod a+rx "/usr/local/bin/protoc" && \
+    cd /  && \
+    rm -rf /protoc
+
+# Install protobuf BigQuery schema generator
+WORKDIR /
+RUN git clone "https://github.com/GoogleCloudPlatform/protoc-gen-bq-schema.git" && \
+    cd "protoc-gen-bq-schema" && \
+    make && \
+    cp "bin/protoc-gen-bq-schema" /usr/local/bin/ && \
+    chmod a+rx "/usr/local/bin/protoc-gen-bq-schema"
+
+# Ensure Google Cloud Platform tools are installed
+RUN gcloud components install \
+    app-engine-go \
+    bq \
+    core \
+    gsutil \
+    app-engine-python \
+    kubectl
+
+# Put wptdashboard code in GOPATH
+RUN ln -s "/wptdashboard" "/go/src/wptdashboard"
+
+# Drop dev environment into source path
+WORKDIR /wptdashboard
```

Comment:
I'm kind of confused about how these files are split up. If the Jenkins runner instances need to send results to BQ using protos, don't they also need the above proto-related things and gcloud installed?

Reply:
Aha! Yes, they will, eventually. Do you want me to upstream more of the installation in to `.base` now or later?